src/s3fs_logger.h

```diff
@@ -230,6 +230,34 @@ void s3fs_low_logprn2(S3fsLog::s3fs_log_level level, int nest, const char* file,
 #define S3FS_PRN_CURL(fmt, ...)   S3FS_LOW_CURLDBG(fmt, ##__VA_ARGS__, "")
 #define S3FS_PRN_CACHE(fp, ...)   S3FS_LOW_CACHE(fp, ##__VA_ARGS__, "")
 
+// Macros to print log with fuse context
+#define PRINT_FUSE_CTX(level, indent, fmt, ...) do {                \
```

Comment:
Could you please replace the `do{...}while(0)` or enclose all the processing including the `fuse_get_context()` call with the following conditional statement inside it?
```
if(S3fsLog::IsS3fsLogLevel(level)){...}
```
This prevents `fuse_get_context` from being called if the debug levels do not match.

Reply:
I did a similar check in FUSE_CTX_DBG, does that work for you? If you prefer I could update PRINT_FUSE_CTX as you suggested.

Reply:
@eryugey I'm sorry for my late reply.

I believe your code will work fine.
However, what I would like to fix is:

I would like `PRINT_FUSE_CTX` to determine the debug level instead of individual `FUSE_CTX_XXXX` macros.
I've also tried the code below and it seems to work fine.
```
#define PRINT_FUSE_CTX(level, indent, fmt, ...) do {                    \
    if(S3fsLog::IsS3fsLogLevel(level)){                                 \
        struct fuse_context *ctx = fuse_get_context();                  \
        if(ctx == NULL){                                                \
            S3FS_LOW_LOGPRN2(level, indent, fmt, ##__VA_ARGS__);        \
        }else{                                                          \
            S3FS_LOW_LOGPRN2(level, indent, fmt"[pid=%u,uid=%u,gid=%u]",\
                ##__VA_ARGS__,                                          \
                (unsigned int)(ctx->pid),                               \
                (unsigned int)(ctx->uid),                               \
                (unsigned int)(ctx->gid));                              \
        }                                                               \
    }                                                                   \
} while (0)

#define FUSE_CTX_INFO(fmt, ...) do {                            \
    PRINT_FUSE_CTX(S3fsLog::LEVEL_INFO, 0, fmt, ##__VA_ARGS__); \
} while (0)

#define FUSE_CTX_INFO1(fmt, ...) do {                           \
    PRINT_FUSE_CTX(S3fsLog::LEVEL_INFO, 1, fmt, ##__VA_ARGS__); \
} while (0)

#define FUSE_CTX_DBG(fmt, ...) do {                             \
    PRINT_FUSE_CTX(S3fsLog::LEVEL_DBG, 0, fmt, ##__VA_ARGS__);  \
} while (0)
```

Reply:
Sure, I'll update as you suggested.

Reply:
@eryugey Thank you for your time. I confirmed the fix.